### PR TITLE
Fix bug where file examples can be corrupted if has multiple extensions

### LINKED ISSUE #4424: Examples bug when filename has two periods
### Describe the bug

Just observed that when attempting to use add an example to a web app with two periods, e.g., `test_data.nii.gz`, the updated filename after example click event corrupted the file extension.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

There is likely a much simpler way to reproduce it, but below is an example that somewhat mimics my application. I am trying to connect the output of the Example click event to be linked to a `gr.File()` widget.

```
import gradio as gr

with gr.Blocks() as demo:
    file_output = gr.File(file_count="single")
    file_output.upload(upload_func(), file_output, file_output)
    
    run_button = gr.Button("Run analysis")
    run_button.click(fn=run_func(), inputs=file_output, outputs=[...])
    
    gr.Examples(examples=["test_data.nii.gz"], inputs=[...], outputs=[...],fn=upload_func(), cache_examples=True)
```

### Screenshot

See updated filename after attempting to use the Example with extension `.nii.gz`.

Screenshot 2023-06-05 at 17 29 25

### Logs

```shell
Not necessary.
```

### System Info

```shell
gradio version: 3.32.0
Browser: Google Chrome
Operating system: Ubuntu Linux 20.04

gradio app deployed through Docker on Hugging Face spaces

The [neukit](https://huggingface.co/spaces/andreped/neukit) gradio app itself can be accessed [here](https://huggingface.co/spaces/andreped/neukit).
```

### Severity

serious, but I can work around it

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Fixes #4424 by preserving the entire filename in the file cached for examples

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Comment by WilliamRoyNelson** [comment]
The changelog for gradio-client is currently a little off. I have a PR to fix it, but I want to make sure that your changes to the changelog don't result in a merge conflict.

Before merging, would you please ensure that everything's in the right place?
https://github.com/gradio-app/gradio/pull/4441

File: test/test_examples.py [hunk 1]
```diff
@@ -372,3 +373,27 @@ async def test_multiple_file_flagging(tmp_path):
 
         assert len(prediction[0]) == 2
         assert all(isinstance(d, dict) for d in prediction[0])
+
+
+@pytest.mark.asyncio
+async def test_examples_keep_all_suffixes(tmp_path):
```
Comment (freddyaboulton) [inline test/test_examples.py:379]: Need to release a new version of the client to get this test to pass

**Comment by abidlabs** [comment]
LGTM @freddyaboulton thanks for this fix! Also fixes a long-standing problem with the name of cached example output files would be different than if you just re-ran the function. 

I'll start by releasing a version of the Python Client (0.2.6) off of this branch. Then, will confirm tests are passing before merging in this PR

**Comment by abidlabs** [comment]
@WilliamRoyNelson I also moved the changelog entries to the right place, thanks for bringing that to our attention

**Comment by abidlabs** [comment]
Everything looks good, thanks folks!

**Comment by freddyaboulton** [comment]
Thanks for the review @abidlabs !

**Pull request merged**